core/src/main/java/hudson/util/RobustMapConverter.java

```diff
@@ -42,13 +45,40 @@
 final class RobustMapConverter extends MapConverter {
     private static final Object ERROR = new Object();
 
+    /**
+     * When available, this field holds the declared type of the keys of the map being deserialized.
+     * This is used to ensure that deserialized keys have the expected type.
+     */
+    // See note above RobustCollectionConverter.elementType for limitations regarding non-concrete parameterized types.
```

Comment:
Does Javadoc work when you have an intervening non-doc comment?

Reply:
As far as I know, yes, for example see https://github.com/openjdk/jdk/blob/2669e22b76c99c1e41a324099154b561e0433b56/src/java.base/share/classes/java/util/HashMap.java#L412-L421. I thought I had seen this pattern used more often in the JDK codebase for implementation notes but that was all I could find in a quick look and it seems to be more typical to use separate block comments. I can adjust the layout.

Reply:
I adjusted things in [801c3e7](https://github.com/jenkinsci/jenkins/pull/9727/commits/801c3e72529775b34cbc77267cfc0ad604cd5bfc).